poller/poller.go

```diff
@@ -0,0 +1,65 @@
+package poller
+
+import (
+	"time"
+
+	"github.com/godaddy/split-go-serializer/api"
+)
+
+// Poller contains cache data and SplitioAPIBinding
+type Poller struct {
+	Cache              int
+	Error              error
+	splitioAPIBinding  api.SplitioAPIBinding
+	pollingRateSeconds int
+	serializeSegments  bool
+	quit               chan bool
+	errorChannel       chan error
+}
+
+// NewPoller returns a new Poller
+func NewPoller(splitioAPIKey string, pollingRateSeconds int, serializeSegments bool) *Poller {
+	if pollingRateSeconds == 0 {
+		pollingRateSeconds = 300
+	}
+	splitioAPIBinding := api.NewSplitioAPIBinding(splitioAPIKey, "")
+	return &Poller{0, nil, *splitioAPIBinding, pollingRateSeconds, serializeSegments, make(chan bool), make(chan error)}
+}
+
+// pollForChanges will get the latest data of splits and segment
+func (poller *Poller) pollForChanges() {
+	// TODO: call getSplits and getSegments to formulate the cach
+	// if any of the returned splits/segments have error:
+	// 1. pass the error to poller.Error and log the error
+	// 2. send the error to poller.errorChannel so it will stop the loop
+	poller.Cache++
+}
+
+// Start creates a goroutine and keep tracking until it stops
+func (poller *Poller) Start() {
+	poller.Error = nil
+	poller.pollForChanges()
+	go poller.jobs()
+}
+
+// Stop sets quit to true in order to stop the loop
+func (poller *Poller) Stop() {
+	poller.quit <- true
+}
+
+// jobs controls whether keep or stop running
+func (poller *Poller) jobs() {
+	ticker := time.NewTicker(time.Duration(poller.pollingRateSeconds) * time.Second)
+	for {
+		select {
+		case <-poller.quit:
+			ticker.Stop()
+			return
+		case <-poller.errorChannel:
```

Comment:
I don't think we want to stop if there's an error, do we?  instead we want a way to surface that error to the caller so that they can handle it however they want (e.g., log it using their logger of choice)

It might be easier to add the error handling later on after we've implemented `pollForChanges`

Reply:
oh okay, I thought we should stop it once we get the error otherwise client will keep getting that error. but yeah probably after a retry the error will be gone.

I will delete that condition.